common/djangoapps/embargo/models.py

```diff
@@ -196,19 +243,104 @@ class CountryAccessRule(models.Model):
         help_text=ugettext_lazy(u"The country to which this rule applies.")
     )
 
+    @classmethod
+    def cache_key_for_consolidated_countries(cls, course_id):
+        """
+        Args:
+            course_id (str): course_id to look for
+        Returns:
+            Consolidated list of accessible countries for given course
+        """
+        return "{}/allowed/countries".format(course_id)
+
+    @classmethod
+    def check_country_access(cls, course_id, country):
+        """
+        Check if the country is either in whitelist or blacklist of countries for the course_id
+
+        Args:
+            course_id (str): course_id to look for
+            country (str): A 2 characters code of country
+
+        Returns:
+            Boolean
+            True if country found in allowed country
+            otherwise check given country exists in list
+        """
+        allowed_countries = cache.get(cls.cache_key_for_consolidated_countries(course_id))
+        if not allowed_countries:
+            allowed_countries = cls._get_country_access_list(course_id)
+            cache.set(cls.cache_key_for_consolidated_countries(course_id), allowed_countries)
+
+        if country == '' or country in allowed_countries:
```

Comment:
These three lines could be reduced to a single line
return country == '' or country in allowed_countries